### PR TITLE
feature: activity item store

### PR DESCRIPTION
**Pull request by moiskillnadne**

### CODE REVIEW

File: src/entities/activity/model/activity.slice.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  itemAnswers: null,
+}
+
+const activitySlice = createSlice({
+  name: "activity",
+  initialState,
+  reducers: {
+    saveActivityDetails: (state, action: PayloadAction<ActivityDetailsState>) => {
+      state.activityDetails = action.payload
+      return state
```
Comment (BamMironov) [inline src/entities/activity/model/activity.slice.ts:52]: No need to return state

File: src/widgets/ActivityDetails/ui/ActivityDetails.tsx [hunk 2]
```diff
@@ -20,6 +22,14 @@ export const ActivityDetailsWidget = (props: ActivityDetailsWidgetProps) => {
     activityId: "30", // Hardcoded
   })
 
+  const { setActivityDetails } = activityModel.hooks.useActivityDetailsState()
+
+  useEffect(() => {
+    if (activityDetails) {
+      setActivityDetails(activityDetails)
+    }
+  }, [activityDetails, setActivityDetails])
```
Comment (BamMironov) [inline src/widgets/ActivityDetails/ui/ActivityDetails.tsx:31]: Please use react query's onSuccess callback instead of useEffect

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/widgets/ActivityDetails/ui/ActivityDetails.tsx (hunk 2 above)
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/ui/ActivityDetails.tsx:31]: Done

File: src/widgets/ActivityDetails/lib/useActivityDetails.ts [hunk 3]
```diff
@@ -25,9 +27,17 @@ export const useActivityDetails = ({ appletId, activityId }: UseActivityDetailsP
...
+  }, [activityById?.data?.result])
+
   return {
-    appletDetails: appletModel.appletBuilder.convertToAppletDetails(appletById?.data?.result),
-    activityDetails: activityModel.activityBuilder.convertToActivityDetails(activityById?.data?.result),
+    appletDetails,
+    activityDetails,
     isError: isAppletError || isActivityError,
     isLoading: isAppletLoading || isActivityLoading,
   }
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/widgets/ActivityDetails/lib/useActivityDetails.ts:43]: Done

**Review by BamMironov — APPROVED** [review]

**Pull request merged**